mindsdb/integrations/handlers/utilities/query_utilities/update_query_utilities.py

```diff
@@ -65,3 +66,40 @@ class UPDATEQueryExecutor(BaseQueryExecutor):
           Because all of the records need to be extracted to be passed in as a DataFrame, this class is not very computationally efficient.
           Therefore, DO NOT use this class if the API/SDK that you are using supports updating records in bulk.
     """
+    def __init__(self, df: pd.DataFrame, set_clauses: List[Tuple[Text, Any]], where_conditions: List[List[Text]]):
+        self.df = df
+        self.set_clauses = set_clauses
+        self.where_conditions = where_conditions
+
+    def execute_query(self) -> pd.DataFrame:
+        """
+        Execute the query.
+        """
+        self.execute_where_clause()
+
+        self.execute_set_clauses()
+
+        return self.df
+    
+    def execute_set_clauses(self)  -> None:
```

Comment:
The implementation that you have given here only makes the change in the output `DataFrame` that is produced by MindsDB. It does not persist the changes to the database/application. It is very hard to come up with a generic solution that does this for every database/application out there. Therefore, we will need to remove this function as well.
This has been explained as clearly as possible in the docstring for the class.

Reply:
The docstring says: NOTE: This class DOES NOT update the relevant records of the entity for you, it will simply return the records that need to be updated based on the WHERE conditions.

I am confused. The execute_set_clause returns the output dataframe to the PagesTable.update method (per docstring). The update method then iterates through the returned dataframe and makes post requests via call_mediawiki_api().

Reply:
Hey @Sohlar,
This is what I think. The `UPDATEQueryParser` returns the `WHERE` conditions and `value_to_update` as a dictionary (the key will be the column to be updated and the value (of that key) is the value to be updated).

Now, you can directly act on these within the `update()` method of the `APITable` implementation.

The edit you have made to the `UPDATEQueryExecutor` (specifically to the `execute_set_clauses`) does the following; it takes the `value_to_update` from the `UPDATEQueryParser` and makes the updates in the `DataFrame` containing the data and returns it to the `update()` method of the `APITable`. These updates made to the DataFrame, however, are not persisted to the database (API) at this point.

The `update()` method then takes this `DataFrame`, loops through the rows and makes API calls like you said. 

My question is, why is it necessary to make updates to the `DataFrame` (if they are not persisted to the database) when you already have the `value_to_update`, which is returned from the `UPDATEQueryParser`? This steps seems redundant. 

You should be able to loop through the DataFrame returned by the `UPDATEQueryExecutor` (original implementation) and make the API calls using the `values_to_update`. I hope this makes sense.